### PR TITLE
[TF:TRT] Update test helpers to handle bool tensors and weights

### PR DESCRIPTION
**Pull request by tfeher**
Enable the handling of bool tensors for in TF-TRT test helper subroutines.

### CODE REVIEW

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc [hunk 1]
```diff
@@ -1369,27 +1389,40 @@ class OpConverterTest : public ::testing::Test {
...
           converter_->weight_store_.GetTempWeights(dtype, dims_adap.AsTrtDims())
               .ConsumeValueOrDie();
-      QCHECK_EQ(weights.size_bytes(), sizeof(T) * values.size())
-          << weights.size_bytes() << " vs " << sizeof(T) * values.size();
-      memcpy(weights.GetPointer<int8>(), values.data(), weights.size_bytes());
+
+      if (tf_type == DT_FLOAT) {
+        std::transform(
+            values.begin(), values.end(), weights.GetPointer<float>(),
+            [](const T in_val) -> float { return static_cast<float>(in_val); });
```
Comment (bixia1) [inline tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc:1396]: Shall we put this into a templated function and use it for the few cases?

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc [hunk 2]
```diff
@@ -1369,27 +1389,40 @@ class OpConverterTest : public ::testing::Test {
...
+        LOG(FATAL) << "Cannot create tensor with type "
+                   << DataTypeString(tf_type);
+      }
     }
     TF_EXPECT_OK(
         converter_->AddTensorOrWeights(name, TRT_TensorOrWeights{weights}));
   }
 
-  template <typename T = int32>
+  // Add test weight without specifying tf_type arg. In this case the initial
```
Comment (bixia1) [inline tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc:1420]: s/Add/Adds/

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc [hunk 3]
```diff
@@ -3675,9 +3716,21 @@ TEST_P(OpConverter_FP32_Test, ConvertCombinedNMS) {
...
+  VLOG(2) << "  <<<<< I am in CreateUnaryOp after Scope::NewRootScope()";
   auto input = ops::Placeholder(s.WithOpName("input"), tf_type);
-  return T(s.WithOpName("my_unary"), input).operation.node()->def();
+  VLOG(2) << "  <<<<< I am in CreateUnaryOp after Placeholder";
+  const tensorflow::Operation& operation =
+      T(s.WithOpName("my_unary"), input).operation;
+  const tensorflow::Node* pNode = operation.node();
+  if (!pNode) {
+    VLOG(2) << "  <<<<<  I am in CreateUnaryOp pNode = NULL";
+  }
```
Comment (bixia1) [inline tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc:3729]: Do we really want these "I am in ..." log message?

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc [hunk 4]
```diff
@@ -3675,9 +3716,21 @@ TEST_P(OpConverter_FP32_Test, ConvertCombinedNMS) {
...
+  VLOG(2) << "  <<<<< Operation: num_inputs: " << operation.num_inputs()
+          << "  num_inputs: " << operation.num_inputs();
+  return pNode->def();
+  //  return T(s.WithOpName("my_unary"), input).operation.node()->def();
```
Comment (bixia1) [inline tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc:3733]: No need for this line.

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc [hunk 5]
```diff
@@ -6666,6 +6719,22 @@ TEST_P(OpConverter_FP32_Test, ConvertUnary) {
...
+  if (1) {
+    // Input is weights, should fail.
+    Reset();
+    const NodeDef node_def = CreateUnaryOp<ops::LogicalNot>(DT_BOOL);
+    AddTestTensor("input", {1, 2, 3}, DT_BOOL, input_values);
+    AddTestWeights("input2", {1, 2, 3}, input_values, DT_BOOL);
+    // RunValidationAndConversion(
+    //     node_def, error::UNIMPLEMENTED,
+    //     "The input \"x\" for LogicalNot must be a tensor");
+  }
```
Comment (bixia1) [inline tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc:6735]: Sorry, what do we want to do here?

File: tensorflow/compiler/tf2tensorrt/utils/trt_engine_utils.cc [hunk 6]
```diff
@@ -77,6 +77,34 @@ Status GetTrtBindingShape(const nvinfer1::ICudaEngine* cuda_engine,
   return Status::OK();
 }
 
+// Setup bindings.
```
Comment (bixia1) [inline tensorflow/compiler/tf2tensorrt/utils/trt_engine_utils.cc:80]: s/Setup/Sets up/

**Comment by tfeher** [comment]
Thanks @bixia1 for the review. Note that this is still a work in progress. Apart from fixing the issues that you pointed out, I plan further refactoring of the helper routines. I will let you know once it is ready for review.

**Comment by tfeher** [comment]
Closing this, in favor of an updated version #55177

**Pull request closed without merging**